test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,173 @@
+// Special Methods
+
+/* This primer covers special methods for classes and records */
+
+// Classes and records can have a handful of methods that are specially named
+// and treated differently than other methods.  These methods enable
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Throughout this primer, the methods are defined as secondary methods
+// using the form: ``proc TypeName.methodName() ...``
+//
+// It is also possible to declare these methods within the body of the
+// class or record as primary methods: ``proc methodName() ...``
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
+// for example with the ``new`` keyword. Here we initialize the record with
+// the given size.
+proc R.init(param size: int = 10) {
+  writeln("Initialized an R instance");
+  this.size = size;
+}
+
+var r = new R();
+
+// The ``deinit`` method will deinitialize the record when it leaves scope.
+// If the record contained any unmanaged classes, open files, etc. this
+// method would be the place to delete them or otherwise clean them up.
+proc R.deinit() {
+  writeln("Deinitialized an R instance");
+}
+
+/*
+  The ``this`` Accessor
+  ---------------------
+*/
+
+// The ``this`` method gives the record the ability to be accessed like an
+// array.  Here we use the index to choose a tuple element.
```

Comment:
I think it'd be worth noting here that *all* Chapel method/function calls can be written as myfunction[1,2] or myfunction(1,2) because otherwise people might be confused about the sudden switch to square brackets.